pkg/controller/operators/olm/operator.go

```diff
@@ -1177,46 +1182,46 @@ func (a *Operator) parseStrategiesAndUpdateStatus(csv *v1alpha1.ClusterServiceVe
 	return installer, strategy, previousStrategy
 }
 
-func (a *Operator) crdOwnerConflicts(in *v1alpha1.ClusterServiceVersion, csvsInNamespace map[string]*v1alpha1.ClusterServiceVersion) error {
-	owned := false
+func (a *Operator) crdOwnerConflicts(in *v1alpha1.ClusterServiceVersion, csvs map[string]*v1alpha1.ClusterServiceVersion) error {
 	for _, crd := range in.Spec.CustomResourceDefinitions.Owned {
-		for csvName, csv := range csvsInNamespace {
-			if csvName == in.GetName() {
-				continue
-			}
-			if csv.OwnsCRD(crd.Name) {
-				owned = true
-			}
-			if owned && in.Spec.Replaces == csvName {
-				return nil
+		for name, csv := range csvs {
+			if name != in.GetName() && in.Spec.Replaces != name && csv.OwnsCRD(crd.Name) {
+				return ErrCRDOwnerConflict
 			}
 		}
 	}
-	if owned {
-		return ErrCRDOwnerConflict
-	}
+
 	return nil
 }
 
-func (a *Operator) apiServiceOwnerConflicts(in *v1alpha1.ClusterServiceVersion, csvsInNamespace map[string]*v1alpha1.ClusterServiceVersion) error {
-	owned := false
-	for _, api := range in.Spec.APIServiceDefinitions.Owned {
-		name := fmt.Sprintf("%s.%s", api.Version, api.Group)
-		for csvName, csv := range csvsInNamespace {
-			if csvName == in.GetName() {
-				continue
-			}
-			if csv.OwnsAPIService(name) {
-				owned = true
-			}
-			if owned && in.Spec.Replaces == csvName {
-				return nil
-			}
-		}
+func (a *Operator) apiServiceOwnerConflicts(csv *v1alpha1.ClusterServiceVersion) error {
```

Comment:
why don't we need to check all csvs here?

Reply:
We're using ownerlabels on the `APIService` itself to determine ownership. If an `APIService` that the given CSV provides is already "owned" by anything other than the given CSV or the CSV it replaces, then a conflict exists.